### PR TITLE
Cleanup Snapshot ITs Further

### PR DESCRIPTION
**Pull request by original-brownbear**
Drying up a few more spots and extracting tests that use their own custom plugins to separate
classes to make the giant `DedicatedClusterSnapshotRestoreIT` suite a little more digestible.

### CODE REVIEW

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/packaging-sample-unix

**Comment by original-brownbear** [comment]
@tlrx @fcofdez obviously very low priority on this one :)

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotBrokenSettingsIT.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        assertThat(getSnapshot("test-repo", "test-snap").state(), equalTo(SnapshotState.SUCCESS));
+
+        logger.info("--> change the test persistent setting and break it");
+        setSettingValue.accept("new value 2");
+        assertSettingValue.accept("new value 2");
+        BrokenSettingPlugin.breakSetting();
+
+        logger.info("--> restore snapshot");
+        try {
+            client.admin().cluster().prepareRestoreSnapshot("test-repo", "test-snap").setRestoreGlobalState(true)
```
Comment (tlrx) [inline server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotBrokenSettingsIT.java:70]: I know this test is muted but I'd expect an expectThrows() here.

File: test/framework/src/main/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java [hunk 2]
```diff
@@ -539,17 +539,17 @@ protected SnapshotInfo getSnapshot(String repository, String snapshot) {
         return snapshotInfos.get(0);
     }
 
+    protected static ThreadPoolStats.Stats snapshotThreadPoolStats(final String node) {
+        return StreamSupport.stream(internalCluster().getInstance(ThreadPool.class, node).stats().spliterator(), false)
+                .filter(threadPool -> threadPool.getName().equals(ThreadPool.Names.SNAPSHOT))
+                .findFirst()
+                .orElseThrow(() -> new AssertionError("Failed to find snapshot pool"));
```
Comment (tlrx) [inline test/framework/src/main/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java:546]: Can we add the node name?

**Review by tlrx — APPROVED** [review]
LGTM with two minor comments

**Comment by original-brownbear** [comment]
Thanks Tanguy!

**Pull request merged**